pkg/provisioner/ironic/validatemanagementaccess_test.go

```diff
@@ -892,6 +892,64 @@ func TestValidateManagementAccessMalformedBMCAddress(t *testing.T) {
 	assert.Equal(t, "failed to parse BMC address information: failed to parse BMC address information: parse \"<ipmi://192.168.122.1:6233>\": first path segment in URL cannot contain colon", result.ErrorMessage)
 }
 
+func TestValidateManagementAccessUpdateBMCAddress(t *testing.T) {
```

Comment:
Could you add another test to see what happens if a user also changes the driver part (e.g. from IPMI to Redfish)? (I wonder if we really want to allow that.. probably?)

Reply:
If the user changes the driver part, the node's driver_info field is updated accordingly (with the new values and defaults), but the node's 'driver' and interfaces are left as it is, which can lead to en error state in Ironic if they end up incorrect. 
Whether we should allow this or not is indeed a good question. The user might have to responsible to ensure that all the node fields are correct, or BMO can update the entire node object as part of this change as well.